libsession/src/main/java/org/session/libsession/messaging/jobs/BatchMessageReceiveJob.kt

```diff
@@ -94,12 +94,24 @@ class BatchMessageReceiveJob(
                         threadMap[threadID]!! += parsedParams
                     }
                 } catch (e: Exception) {
-                    Log.e(TAG, "Couldn't receive message.", e)
-                    if (e is MessageReceiver.Error && !e.isRetryable) {
-                        Log.e(TAG, "Message failed permanently",e)
-                    } else {
-                        Log.e(TAG, "Message failed",e)
-                        failures += messageParameters
+                    when (e) {
+                        is MessageReceiver.Error.DuplicateMessage, MessageReceiver.Error.SelfSend -> {
+                            Log.i(TAG, "Couldn't receive message, failed with error: ${e.message}")
+                            failures += messageParameters
```

Comment:
Are duplicates and self-sends in error recoverable and could be tried again or permanent failures?

Reply:
They aren't really errors, they are just an artefact of how the messaging system is designed (on iOS I've actually removed them from the logs because they are essentially spam logs)

I'll remove the `failures` incrementing since if there were only duplicates/self-sends then we wouldn't want the batch job to fail

Reply:
I think the `failures` list will be used to reschedule a new `BatchMessageReceiveJob` that contains only those failed message parameters, in the event of a new closed group message trying to be processed from the group's swarm before we had received the latest encryption keys for instance

Reply:
It doesn't currently appear to modify the list of messages though so if `failures` is greater than `0` it's going to re-process all of the messages in the batch job (including any successful/duplicate/self-send messages) so there's the potential for further improvements here to avoid reprocessing messages we no longer care about